### PR TITLE
Doc/update

### LINKED ISSUE #239: [PE-D] Evidence of addressbook in the app
Upon initial launch of the app, I noticed that at the bottom, there is still a mention of the addressbook.json.

Seeing as your application is no longer an addressbook app but a teaching app, you may want to remove the mention or replace it with a different name.

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: rachel170/ped#2

### LINKED ISSUE #240: [PE-D] Logging Command: Able to accept nusstu\tester
I noticed that entering: edit -l u/tester p/testpassword
Produces the follow error message specifying that the username has to be 'nusstu\' followed by NusNetID.

However, when I enter: edit -l u/nusstu\tester p/testpassword
The value is accepted despite 'tester' not being a valid NusNetID. I am unsure if it is intended or not so I am reporting it just to inform you.

Note: It seems to accepted any string after 'nusstu\'. Additionally, despite being a login function, the "Please login" message does not change on successful command.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: rachel170/ped#3

### LINKED ISSUE #247: [PE-D] Unable to add or delete students through command line
### Expected Behavior

As a teaching assistant, I would like to add or delete students through the command lind interface such that I can introduce or remove students from my course quickly.

### Actual Behavior

I can only have up to 6 students. Using the `edit -s` command, I can change their details. However, there are no commands to add or delete the students.

While it is possible for me to edit the `.json` file and add more students manually, it is not ideal if I want to perform the add and delete operations quickly by typing.

Would it be possible to add the add/delete student commands? Or if this intended, maybe it could be described in the user guide! 

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: w-yeehong/ped#3

### LINKED ISSUE #248: [PE-D] User guide does not describe how to edit a student's name
### Expected Behavior

The user guide should mention `n/NAME` is used to edit a student's name under the `edit -s` section.

### Actual Behavior

The user guide does not include `n/NAME` as a parameter under the `edit -s` section. The user has to rely on
the result box to discover the `n/NAME` parameter.

## Steps to Reproduce

1. Refer to user guide section 3.2.

1. The `n/NAME` parameter has not been included in the format.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: w-yeehong/ped#4

### LINKED ISSUE #249: [PE-D] Unable to add participation score
### Expected Behavior

The `edit -s` command should be able to add participation score to a user.

### Actual Behavior

The `edit -s` command does not add a participation score to a user as `s/PARTICIPATION_SCORE` is not recognized as a valid parameter.

## Steps to Reproduce

1. Without logging in, execute the command `edit -s 2 t/maryjane132 s/3`.

1. `s/3` is treated as part of the Telegram username instead of the participation score.

1. Again, without logging in, execute the command `edit -s 2 s/3`.

1. The `s/3` argument on its own results in an error message `Invalid command format!`.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: w-yeehong/ped#5

### LINKED ISSUE #250: [PE-D] No error message when failing to log in
### Expected Behavior

If a user enters incorrect credentials when using the `edit -l` command to log in, an error message should be displayed in the user interface.

### Actual Behavior

The user interface continues to prompt the user to log in if he enters incorrect credentials. An error message can be seen through the logs when restarting the application.

## Steps to Reproduce

1. Execute the command `edit -l u/nusstu\a p/testpassword`. 

1. The following output results.

3. Notice that the user is not informed that the credentials are incorrect and the prompt to log in remains.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: w-yeehong/ped#6

### LINKED ISSUE #266: [PE-D] View -p does not work as expected
Build Number: Version Number 1.3
Reported By: Chu Yi-Ting
Reported On: 2020-10-30
Reason: Defect
Severity: High
Environment: macOS version 10.15.7

Description:
view -p should show the studio partition marks of all the students of the logged in avenger (the user guide does not specify clearly)

Expected Behavior: 
list the partition marks of all the students or throw error message which asks the user to specify the student name.

Actual Behavior: 
Flag provide is invalid.

Steps To Reproduce:
1.	Log into an avenger account
2.     Type "view -p"

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: chuyiting/ped#16

### LINKED ISSUE #267: [PE-D] Log-status not updated when switching account
Build Number: Version Number 1.3
Reported By: Chu Yi-Ting
Reported On: 2020-10-30
Reason: Defect
Severity: High
Environment: macOS version 10.15.7

Description:
edit -l u/... p/, expects to switch account (User Guide does not specify this, but the phrasing might make the user think this way). However, the data is not changed accordingly.

Expected Behavior: 
The missions and the quests are updated according to the log-in status which is the current username and passwork.

Actual Behavior: 
The username and the password are updated, but the account is not.

Steps To Reproduce:
1.	Log in with a non-avenger account.
2.     Log in with an avenger account.
Note: reversing the sequence also generates similar outcome.

This picture can be generated after first log-in to a non-avenger account and then log-in with an avenger account.

Note: currently only the first edit command would incur log-in process, if this is desirable,  maybe you can include that in the user guide. However, if the user has typo when first calls the edit command, then the user would need to relaunch the application to log-in correctly. Maybe, this can be a bit user-friendly. You can think about this and come up with a more user-friendly design, but I really like your ideas of integration with 1101S, cool idea, keep it up.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: chuyiting/ped#17

### PR DESCRIPTION
**Pull request by peironggg**
Fixes #267 
Fixes #250 
Fixes #249 
Fixes #248 
Fixes #247 
Fixes #240 
Fixes #239 
Fixes #266

### CODE REVIEW

**Review by ngzhenteng — COMMENTED** [review]
Thanks for fixing the bugs, LGTM!

**Review by wilinetan — COMMENTED** [review]
Thanks for fixing the bus and changing all the addressbook references to jarvis! 👍

File: docs/UserGuide.md [hunk 1]
```diff
@@ -44,8 +44,7 @@ Jarvis is a desktop app for CS1101S Teaching Assistants (Avengers), optimized fo
         * [6.6.9 Viewing deadline for missions: `view -m`](#669-viewing-deadline-for-missions---m)
         * [6.6.10 Viewing deadline for quests: `view -q`](#6610-viewing-deadline-for-quests---q)
         * [6.6.11 Viewing ungraded missions: `view -um`](#6611-viewing-ungraded-missions---um)
-        * [6.6.12 Viewing ungraded quests: `view -uq`](#6612-viewing-ungraded-quests---uq)
-        * [6.6.13 Viewing studio participation marks of a specific student: `view -p`](#6613-viewing-studio-participation-marks-of-a-specific-student--view--p)
+        * [6.6.12 Viewing ungraded quests: `view -uq`](#6612-viewing-ungraded-quests---uq)
```
Comment (wilinetan) [inline docs/UserGuide.md:48]: the numbering seems to off from 6.6.12 to 6.6.14

File: docs/UserGuide.md [hunk 2]
```diff
@@ -271,14 +270,21 @@ Adds a `Mastery Check` session with a `Student` at a specific `date` and `time`.
...
 Similar to the format for editing students above, you can use tags to specify the field to be edited.
-<br>**Format: `edit -l u/LUMINUS_USERNAME p/LUMINUS_PASSWORD`
+<br>**Format: `edit -l u/LUMINUS_USERNAME p/LUMINUS_PASSWORD`**
 <br>Examples:
 * `edit -l u/nusstu\e1234567 p/testpassword`
 * `edit -l p/testpassword`
 
+Do note that it takes a load time of around ~5 seconds after entering this command for the changes from Sourceacademy to be reflected in the GUI.
+
+#### 6.4.2 Editing Student Informaion : `-s`
```
Comment (wilinetan) [inline docs/UserGuide.md]: Is this inserted in the content page as well?

File: src/main/java/seedu/jarvis/model/greeting/Greeting.java [hunk 3]
```diff
@@ -24,14 +24,19 @@ public StringProperty getGreeting() {
...
             this.greeting.setValue(DEFAULT_GREETING);
         } else {
             this.greeting.setValue(padWithIntro(newGreeting));
         }
 
     }
 
+    private boolean verifyGreeting(String greeting) {
+        return greeting == null || greeting.isEmpty() || greeting.equals(DEFAULT_GREETING)
+                || greeting.contains("Welcome");
```
Comment (wilinetan) [inline src/main/java/seedu/jarvis/model/greeting/Greeting.java:37]: Does this mean that the greeting will always be reset upon start up?

**Review by wilinetan — COMMENTED** [review]
Thanks for fixing the bugs and refactoring address to jarvis!

**Comment by peironggg** [comment]
> Thanks for fixing the bus and changing all the addressbook references to jarvis! 👍

bus

File: src/main/java/seedu/jarvis/model/greeting/Greeting.java (hunk 3 above)
Comment (peironggg) [reply src/main/java/seedu/jarvis/model/greeting/Greeting.java:37]: yes, good too cos if ppl change user, the greeting should change as well.

File: docs/UserGuide.md (hunk 2 above)
Comment (peironggg) [reply docs/UserGuide.md]: nope it is not, let me insert it. Good catch 👍

**Pull request merged**